android/src/main/java/com/mattermost/networkclient/helpers/Helper.kt

```diff
@@ -199,3 +244,40 @@ fun formUrlString(baseUrl: String, endpoint: String): String{
 fun emitEvent(reactContext: ReactContext, eventName: String, params: Any){
     reactContext.getJSModule(RCTDeviceEventEmitter::class.java).emit(eventName, params)
 }
+
+/**
+ * Get's the certificates from a P12 File, and saves them in the AndroidKeyStore
+ *
+ * @param pathToFile
+ * @param password
+ */
+private fun getAndSaveCertsFromP12(pathToFile: String, password: String): Enumeration<String> {
+    val aliases: Enumeration<String>;
+    try {
+        // Get an instance of the AndroidKeyStore
+        val androidKeyStore: KeyStore = KeyStore.getInstance("AndroidKeyStore")
+        androidKeyStore.load(null)
+
+        // Create an instance of a pkcs12 keystore
+        val keyStore: KeyStore = KeyStore.getInstance("pkcs12")
+
+        // Load our certificate file into it, with the password
+        keyStore.load(FileInputStream(pathToFile), password.toCharArray())
+
+        // Get all the key aliases
+        aliases = keyStore.aliases()
+
+        // Loop through them
+        while (aliases.hasMoreElements()) {
+            val alias = aliases.nextElement() as String
+            val cert: X509Certificate = keyStore.getCertificate(alias) as X509Certificate
+
+            // Add the certificate to the KeyStore
+            androidKeyStore.setCertificateEntry(alias, cert)
```

Comment:
Do we use any flags to save/retrieve from the store?

Like: should the device be unlocked? Or can this be access at all times? I guess no hardware cryptography is needed, so for software do you know which one is used?

@migbot Same question I guess applies to iOS and the keychain

Reply:
> @enahum I used [kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly](https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlockthisdeviceonly) for accessing both token and cert from the Keychain.

So backup and restore across devices will require us to import the cert again, yes?

And things like background tasks or fetching notification content + ack will only be possible after the first unlock.

This one allows backup https://developer.apple.com/documentation/security/ksecattraccessibleafterfirstunlock

Obviously always allowing access is now deprecated, so I think we should document this, an be opened to someone contributing this value as a setting